Made Room component

DmBorozdin opened the pull request:
The Room component is made.
Changed Card component. Made a redirect to the page with detailed information when clicking on the card. Made getting one unit of information on the page with detailed information using RTKQuery. Added a selector to get the list of favorites as an object. Added reducer to add to favorites from offer page.

### Что сделано

- Сделан Room компонент
- Изменён компонент Card. При нажатии на карточку происходит переход на страницу с детальной информацией.
- Настроена загрузка детальной информации для выбранного отеля с помощью RTKQuery.
- Добавлен селектор для получения списка избранного в виде объекта для отображения индикатора Избранного.
- Добавлен reducer для добавление в избранное отеля со страницы с детальной информации.

### Чек-лист

- [x] ПР имеет четкое **название**, короткое и описывающее его суть.
- [x] ПР имеет четкое **описание** того, что в рамках него было сделано.
- [x] Я провел **само-ревью** ПРа. Там нет кода, который я забыл удалить, случайно закоммитил и т.п.
- [x] ПР не содержит кода, который не соответствует цели ПРа.

DmBorozdin committed:
The Room component is made.
Changed Card component. Made a redirect to the page with detailed information when clicking on the card.
Made getting one unit of information on the page with detailed information using RTKQuery.
Added a selector to get the list of favorites as an object.
Added reducer to add to favorites from offer page.

VictorDanilov commented on src/components/room/room.tsx:
```diff
@@ -1,14 +1,216 @@
+import { addToFavoriteFromDetailsPage } from "../../store/users/users";
+import { useAppSelector } from "../../app/hooks";
+
+const getAssessmentDescription = (rating: number) => {
```
в отдельный файл

VictorDanilov commented on src/components/room/room.tsx:
```diff
@@ -1,14 +1,216 @@
+    data: offer,
+    isLoading,
+    isSuccess,
+  } = useGetOfferQuery(pageId.toString());
```
Зачем переводить pageId в намбер, чтобы потом тут и ниже в строку переводить опять?

VictorDanilov commented on src/components/room/room.tsx:
```diff
@@ -1,14 +1,216 @@
+                  </ul>
+                </div>
+
+                <section className={styles.reviews}>
```
резать файл этот потом на отдельные компоненты нужно

VictorDanilov commented on src/services/apiSlice.ts:
```diff
@@ -0,0 +1,105 @@
+    };
+  response.data.amenitiesScreen.forEach(
+    (element: { title: string; content: string[] }) => {
+      if (
+        element.title === "Comfort" ||
+        element.title === "Room features" ||
+        element.title === "Kitchen" ||
+        element.title === "Entertainment" ||
+        element.title === "Bath"
+      ) {
```
Готовь массив текстовок за компонентом. Тут используй массив.includes(element.title)

VictorDanilov commented on src/services/apiSlice.ts:
```diff
@@ -0,0 +1,105 @@
+  endpoints: (builder) => ({
+    getOffer: builder.query({
+      query: (offerId: string) =>
+        `/getHotelDetails?id=${offerId}&checkIn=${dayjs().format(
+          DATE_FORMAT
+        )}&checkOut=${dayjs()
+          .add(1, "d")
+          .format(DATE_FORMAT)}&currencyCode=USD`,
```
через params задавай квери параметры. Сложно читать.

VictorDanilov commented on src/store/users/selector.ts:
```diff
@@ -12,3 +12,13 @@ export const getFavorites = createSelector(
+    const favorites = users.users.find(
+      (user) => user.id === users.authUser
+    )?.favorites;
+    return favorites ? favorites : {};
```
```suggestion
    return favorites || {};
```

DmBorozdin committed:
Bugs fixed.
Removed reviews and amenities from the room component.
API key moved to env file
